src/unix_socket.cpp

```diff
@@ -0,0 +1,39 @@
+#include "unix_socket.h"
+#include <chrono>
+#include <thread>
+
+UnixSocketServer::UnixSocketServer() {
+    std::cout << "Creating Unix socket for comms with python interface...\n";
+    sock = socket(AF_UNIX, SOCK_DGRAM, 0);
+    if(sock == -1) {
+        perror("socket");
+        exit(EXIT_FAILURE);
+    }
+    memset(&server_addr, 0, sizeof(struct sockaddr_un));
+    server_addr.sun_family = AF_UNIX;
+    strncpy(server_addr.sun_path, "/tmp/snr_socket.sock", sizeof(server_addr.sun_path) - 1);
+    
+    // Remove old socket file if it exists
+    unlink(server_addr.sun_path);
+    if(bind(sock, (struct sockaddr *)&server_addr, sizeof(struct sockaddr_un)) == -1) {
+        perror("bind");
+        exit(EXIT_FAILURE);
+    }
+}
+void UnixSocketServer::sendMessage(const char *message) {
+    int len = strlen(message) + 1;
+    int bytes_sent = sendto(sock, message, len, 0, (struct sockaddr *)&server_addr, sizeof(struct sockaddr_un));
```

Comment:
You should not be using `strlen` here. The underlying data that you're trying to send is not a string and issues are probably occurring when the strlen finds the a terminating null-character in the data that your casting to a char pointer above before it gets to the actual end of the gps status data.

[](https://cplusplus.com/reference/cstring/strlen/)

```suggestion
void UnixSocketServer::sendMessage(const void *msg , size_t len) {
    
    int bytes_sent = sendto(sock, message, len, 0, (struct sockaddr *)&server_addr, sizeof(struct sockaddr_un));
```

I will add a suggestion above for how I think you should cast the data.